src/components/Products/Products.jsx

```diff
@@ -0,0 +1,45 @@
+import { useEffect, useState } from "react";
+import axios from 'axios';
+import Pagination from "./Pagination/Pagination";
+import Product from "./Product/Product";
+
+const Products = () => {
+  const [products, setProducts] = useState([]);
+  const [totalProducts, setTotalProducts] = useState();
+  const [currentPage, setCurrentPage] = useState(1);
+  const [productsPerPage, setProductsPerPage] = useState(16);
+
+  const pages = [];
+  for(let i = 1; i < Math.ceil(totalProducts / productsPerPage); i++){
+    pages.push(i);
+  }
+
+
+  useEffect(() => {
+    const fetchData = async () => {
+      const response = await axios.get(`http://localhost:3000/products?_page=${currentPage}&_limit=${productsPerPage}`);
```

Comment:
```suggestion
      const response = await axios.get('http://localhost:3000/products', 
      {
         params: {
              _page: currentPage,
              _limit: productsPerPage
              ...(name_like && { name_like: name_like })
         }
      })
```
Có thể code đơn giản và chặt chẽ hơn như này. Trong trường hợp muốn  có thêm bớt params gì dựa theo condition thì nó cũng sẽ dễ dàng hơn, ví dụ như muốn thêm params name_like